src/parser/person.rs

```diff
@@ -31,17 +33,16 @@ pub fn consume<R: Read>(context: &mut Context<R>, tagname: &'static str) -> Resu
                 "email" => person.email = Some(email::consume(context)?),
                 "link" => person.link = Some(link::consume(context)?),
                 child => {
-                    bail!(ErrorKind::InvalidChildElement(
-                        String::from(child),
-                        "person"
-                    ));
+                    return Err(GpxError::InvalidChildElement(String::from(child), "person"));
                 }
             },
             XmlEvent::EndElement { ref name } => {
-                ensure!(
-                    name.local_name == tagname,
-                    ErrorKind::InvalidClosingTag(name.local_name.clone(), "person")
-                );
+                if name.local_name != tagname {
```

Comment:
heh, that's a dumb example because you could just do:
```
if name.local_name != tagname {
    return Err(GpxError::InvalidClosingTag(name.local_name.clone(), "person"));
}
```

But my point stands: "special syntax for guard statements is cool".

Reply:
Yeah, the `error_chain` version was much better. Gonna leave a "revisit this when 3137 has been stable for a few releases" TODO, since we don't have an official MSRV policy…